### ISSUE TITLE
Lite client verification error in all usage: insufficient power

### ISSUE THREAD

**Issue by cwgoes** [tendermint/tendermint#2862]
## Summary of Bug 
`gaiacli query *` is returning error for validators who have already bonded.

## Steps to Reproduce
- Join a node to the `gaia-9000` testnet
- `gaiacli tx create-validator` to create your validator
- Run `gaiacli query signing-info $(gaiad tendermint show-validator) --chain-id gaia-9000`
- See the error:
```
ERROR: Invalid commit -- insufficient old voting power: got 200, needed 267
```

Moved from https://github.com/cosmos/cosmos-sdk/issues/2756, cc @jackzampolin.

**Comment by cwgoes** [comment]
The relevant error is in `types/validator_set.go` - https://github.com/tendermint/tendermint/blob/master/types/validator_set.go#L306.

**Comment by melekes** [comment]
Not immediately obvious why the commit does not have enough voting power. Also, there's no info about commit itself 😞

**Comment by HaoyangLiu** [comment]
@cwgoes 
Do you have located the root of this issue?

Now in tendermint lite client, when validatorset change, the verification algorithm requires the precommits have more than 2/3 voting power in both old validator set and new validator set. However, if the validator set change is much significant, the lite client verification will be failed. 

Currently, we already have `nextValidatorHash`. If the hash of new validator set equals to `nextValidatorHash`, and the precommit has more than 2/3 voting power in new validator set, then the block commit is supposed to be valid. It is not necessary to change the voting power again old validator set.

What do you think about my opinion?

**Comment by ebuchman** [comment]
The problem is this: https://github.com/tendermint/tendermint/blob/v0.28.1/lite/dynamic_verifier.go#L234

But that error never gets returned by VerifyFutureCommit, so instead of proceeding with the bisection to find a commit we can verify, it looks like we just fail.

It shouldn't be hard to fix. The more concerning thing is the lack of testing. I should have a PR up shortly

**Comment by ebuchman** [comment]
I think https://github.com/tendermint/tendermint/pull/3171 should fix in. We need to add much more testing for this package.

**Comment by HaoyangLiu** [comment]
@ebuchman 
I have add a [comment](https://github.com/tendermint/tendermint/pull/3171/files#r249307540) to your PR.

**Comment by ebuchman** [comment]
Fixed in #3171. Will ship in v0.29

**Comment by jackzampolin** [comment]
This is awesome @ebuchman !! Thank you for the fix!!!
